nodejs/week3/homework/warmup/index.js

```diff
@@ -3,4 +3,145 @@ const app = express();
 
 app.get("/", (req, res) => res.send("nodejs week3 homework"));
 
-app.listen(3000, () => console.log(`Calculator:listening on port 3000`));
+//addition
+//with get
+app.get("/numbers/add", (req, res) => {
+  const first = parseInt(req.query.first);
+  const second = req.query.second;
+  let sum = first;
+  if (Array.isArray(req.query.second)) {
+    for (let i = 0; i < second.length; i++) {
+      sum += parseInt(second[i]);
+    }
+  } else {
+    sum += parseInt(second);
+  }
+
+  res.json({ sum });
+});
+// with post
+app.post("/numbers/add", (req, res) => {
+  const first = parseInt(req.query.first);
+  const second = req.query.second;
```

Comment:
Not exactly!
Since this is a POST request, it usually means that the parameters will be inside the request's body and not in the query. So, you need to read them from there.
Here's a useful link that tells you how to do that:
https://nodejs.dev/learn/get-http-request-body-data-using-nodejs
And this is a sample of what you need to do:
```
app.use(
  express.urlencoded({
    extended: true
  })
)
app.post('/numbers/add', (req, res) => {
  const values = Object.values(req.body);
})
```